### PR TITLE
Properly export function defined in test which uses global_asm!()

### PR DESCRIPTION
**Pull request by bjorn3**
Currently the test passes with the LLVM backend as the codegen unit partitioning logic happens to place both the global_asm!() and the function which calls the function defined by the global_asm!() in the same CGU. With the Cranelift backend it breaks however as it will place all assembly in separate codegen units to be passed to an external linker.

### CODE REVIEW

File: tests/ui/asm/x86_64/issue-96797.rs [hunk 1]
```diff
@@ -11,7 +11,14 @@ use std::arch::global_asm;
 #[no_mangle]
 fn my_func() {}
 
-global_asm!("call_foobar: jmp {}", sym foobar);
+global_asm!("
+.globl call_foobar
```
Comment (bjorn3) [inline tests/ui/asm/x86_64/issue-96797.rs:15]: This is the actual change that matters. The rest is more to match what rustc would produce. (a separate section for each function and proper type and size metadata for the symbol.)

File: tests/ui/asm/x86_64/issue-96797.rs [hunk 2]
```diff
@@ -11,7 +11,14 @@ use std::arch::global_asm;
...
+.type call_foobar,@function
+.section .text.call_foobar,\"ax\",@progbits
+call_foobar: jmp {}
+.size call_foobar, .-call_foobar
```
Comment (workingjubilee) [inline tests/ui/asm/x86_64/issue-96797.rs:19]: Hm. What does this `.-call_foobar` expression mean?

Comment (asquared31415) [reply tests/ui/asm/x86_64/issue-96797.rs:19]: `.` (current position) minus the position of the symbol `call_foobar`

File: tests/ui/asm/x86_64/issue-96797.rs [hunk 3]
```diff
@@ -11,7 +11,14 @@ use std::arch::global_asm;
...
+.text
```
Comment (asquared31415) [inline tests/ui/asm/x86_64/issue-96797.rs:20]: Should this instead be using `.pushsection` and `.popsection` to make sure that the section is not modified by starting in something other than plain `.text`

Comment (bjorn3) [reply tests/ui/asm/x86_64/issue-96797.rs]: Done

**Pull request merged**